x/authority/types/message_update_chain_info_test.go

```diff
@@ -0,0 +1,109 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+func TestMsgUpdateChainInfo_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name string
+		msg  *types.MsgUpdateChainInfo
+		err  error
+	}{
+		{
+			name: "valid message",
+			msg:  types.NewMsgUpdateChainInfo(sample.AccAddress(), sample.ChainInfo(42)),
+		},
+		{
+			name: "invalid creator address",
+			msg:  types.NewMsgUpdateChainInfo("invalid", sample.ChainInfo(42)),
+			err:  sdkerrors.ErrInvalidAddress,
+		},
+		{
+			name: "invalid chain info",
+			msg: types.NewMsgUpdateChainInfo(sample.AccAddress(), types.ChainInfo{
+				Chains: []chains.Chain{
+					{
+						ChainId:     0,
+						ChainName:   chains.ChainName_empty,
+						Network:     chains.Network_optimism,
+						NetworkType: chains.NetworkType_testnet,
+						Vm:          chains.Vm_evm,
+						Consensus:   chains.Consensus_op_stack,
+						IsExternal:  true,
+					},
+				},
+			}),
+			err: sdkerrors.ErrInvalidRequest,
```

Comment:
NIT: it would also be good to assert the error message "chain ID must be positive". Not required though.